client/ctclient/ctclient.go

```diff
@@ -216,6 +219,63 @@ func getInclusionProofForHash(ctx context.Context, logClient *client.LogClient,
 	}
 }
 
+func getConsistencyProof(ctx context.Context, logClient *client.LogClient) {
+	if *treeSize <= 0 {
+		log.Fatal("No valid --size supplied")
+	}
+	if *prevTreeSize <= 0 {
+		log.Fatal("No valid --prev_size supplied")
+	}
+	var hash1, hash2 []byte
+	if *prevHash != "" {
+		var err error
+		hash1, err = hex.DecodeString(*prevHash)
+		if len(hash1) != 32 {
+			err = fmt.Errorf("invalid hash length %d", len(hash1))
+		}
+		if err != nil {
+			log.Fatalf("Invalid --prev_hash: %v", err)
+		}
+	}
+	if *treeHash != "" {
+		var err error
+		hash2, err = hex.DecodeString(*treeHash)
+		if len(hash1) != 32 {
+			err = fmt.Errorf("invalid hash length %d", len(hash2))
+		}
+		if err != nil {
+			log.Fatalf("Invalid --tree_hash: %v", err)
+		}
+	}
+	getConsistencyProofBetween(ctx, logClient, *prevTreeSize, *treeSize, hash1, hash2)
+}
+
+func getConsistencyProofBetween(ctx context.Context, logClient *client.LogClient, first, second int64, prevHash, treeHash []byte) {
+	proof, err := logClient.GetSTHConsistency(ctx, uint64(first), uint64(second))
+	if err != nil {
+		if err, ok := err.(client.RspError); ok {
+			log.Fatalf("get-sth-consistency failed: %q, detail:\n  %s", err, string(err.Body))
+		}
+		log.Fatalf("Failed to get-sth-consistency: %v", err)
+	}
+	fmt.Printf("Consistency proof from size %d to size %d:\n", first, second)
+	for _, e := range proof {
+		fmt.Printf("  %x\n", e)
+	}
+	if prevHash == nil || treeHash == nil {
+		return
```

Comment:
Maybe it should complain if one of the hashes is nil to let the user know it needs both.